components/AutoComplete/AutoComplete.jsx

```diff
@@ -0,0 +1,381 @@
+    if (selectedItem) selectedItem(target.value);
+    setCursor(0);
+    handleFilter(target.value);
+  };
+
+  /* istanbul ignore next */
+  const handleClickOutside = event => {
+    if (!wrapperRef.current.contains(event.target)) {
+      setShowSuggestions(false);
+    }
+  };
+
+  const handleInputClick = () => {
+    const filteredItem = filterItens(userTypedValue);
+    setFilterSuggestions(filteredItem);
+    setFilterSuggestionsLength(filteredItem.length);
+    setShowSuggestions(true);
+  };
+
+  const handleItemClick = item => {
+    setUserTypedValue(item);
+    if (selectedItem) selectedItem(item);
+    setShowSuggestions(false);
+  };
+
+  const handleClearValue = () => {
+    setUserTypedValue('');
+  };
+
+  /* istanbul ignore next */
+  const handleEscPress = ({ key }) => {
+    if (key === EscapeKeyPressValue) {
+      setShowSuggestions(false);
+      wrapperRef.current.children[1].focus();
+      setCursor(0);
+    }
+  };
+
+  const generateSuggestions = () =>
+    showSuggestions && filterSuggestions ? (
+      <List
+        theme={theme}
+        role="listbox"
+        id="autocompleteOptions"
+        ref={listOptions}
+      >
+        {filterSuggestions.map((item, index) => (
+          <ListItens
+            key={item}
+            aria-posinset={index}
+            onClick={() => handleItemClick(item)}
+            theme={theme}
+            aria-selected={index === cursor}
+            role="option"
+            tabIndex="-1"
+          >
+            {item}
+          </ListItens>
+        ))}
+      </List>
+    ) : null;
+
+  const generateAssistiveDescript = () => {
+    if (showSuggestions) {
+      return assistiveDescriptionDropDownOpen;
+    }
+    return assistiveDescriptionDefault;
+  };
+
+  const downPress = useKeyPress('ArrowDown');
+  const upPress = useKeyPress('ArrowUp');
+  const enterPress = useKeyPress('Enter');
+  const tabPress = useKeyPress('Tab');
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && filterSuggestionsLength && downPress) {
+      const selectedCursor =
+        cursor < filterSuggestionsLength - 1 ? cursor + 1 : cursor;
+      setCursor(selectedCursor);
+      listOptions.current.children[selectedCursor].focus();
+    }
+  }, [downPress]);
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && filterSuggestionsLength && upPress) {
+      const selectedCursor = cursor > 0 ? cursor - 1 : cursor;
+      setCursor(selectedCursor);
+      listOptions.current.children[selectedCursor].focus();
+    }
+    if (upPress && cursor === 0) {
+      wrapperRef.current.children[1].focus();
+      setCursor(-1);
+    }
+  }, [upPress]);
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && filterSuggestionsLength && enterPress) {
+      setUserTypedValue(filterSuggestions[cursor]);
+      setShowSuggestions(false);
+      wrapperRef.current.children[1].focus();
+    }
+  }, [cursor, enterPress]);
```

Comment:
Shouldn't the EnterPress open options too?

Reply:
You mean the enterPress open the options? If it is about that, for what I searched, none of them would open the options with enter